nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2294,6 +2297,36 @@ async fn instance_serial_console(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Connect to an instance's serial console
+#[channel {
+    protocol = WEBSOCKETS,
+    path = "/organizations/{organization_name}/projects/{project_name}/instances/{instance_name}/serial-console/stream",
```

Comment:
Is the idea we'll have both the streaming and non-streaming endpoint in the limit? I assume the latter for the buffer playback behaviour?

Reply:
the streaming endpoint will eventually get params to start from an arbitrary byte offset (within the buffer) as well once i've moved the longer buffering into propolis-server, at which point the main reasons to keep the non-streaming one around would be
- wanting to separate out permission to read it vs. read/write
- wanting to provide an API for accessing it that doesn't *require* websocket support